pkg/controllers/override/overridepolicy_controller.go

```diff
@@ -176,32 +176,29 @@ func (c *Controller) enqueueFederatedObjectsForFTC(ftc *fedcorev1a1.FederatedTyp
 	logger.V(2).Info("Enqueue federated objects for FTC")
 
 	allObjects := []fedcorev1a1.GenericFederatedObject{}
-	fedObjects, err := c.fedObjectInformer.Lister().List(labels.Everything())
-	if err != nil {
-		c.logger.Error(err, "Failed to enqueue FederatedObjects for override policy")
-		return
-	}
-	for _, obj := range fedObjects {
-		allObjects = append(allObjects, obj)
-	}
-	clusterFedObjects, err := c.clusterFedObjectInformer.Lister().List(labels.Everything())
-	if err != nil {
-		c.logger.Error(err, "Failed to enqueue ClusterFederatedObjects for override policy")
-		return
-	}
-	for _, obj := range clusterFedObjects {
-		allObjects = append(allObjects, obj)
-	}
-
-	for _, obj := range allObjects {
-		sourceMetadata, err := obj.GetSpec().GetTemplateMetadata()
+	labelsSet := labels.Set{ftc.GetSourceTypeGVK().GroupVersion().String(): ftc.GetSourceTypeGVK().Kind}
```

Comment:
Can we extract this common part into `/pkg/util/fedobjectadapters`? eg: ListAllFedobjsForFTC(ftc, foLister,cfoLister,labelSelector)

Reply:
I think it is not necessary because there's only two lines.

Reply:
The code has been updated.